extension/commands/importNodesToEnvironmentCommand.ts

```diff
@@ -88,24 +92,44 @@ export async function importNodesToEnvironment(environmentRegistryEntry: FabricE
                         nodes = [nodes];
                     }
 
-                    nodesToUodate.push(...nodes);
+                    nodesToUpdate.push(...nodes);
                 } catch (error) {
                     addedAllNodes = false;
                     outputAdapter.log(LogType.ERROR, `Error importing node file ${nodeUri.fsPath}: ${error.message}`, `Error importing node file ${nodeUri.fsPath}: ${error.toString()}`);
                 }
             }
         } else {
 
+            const keytar: any = getCoreNodeModule('keytar');
+            if (!keytar) {
+                throw new Error('Error importing the keytar module');
+            }
+
             const GET_ALL_COMPONENTS: string = '/ak/api/v1/components';
             const url: string = await UserInputUtil.showInputBox('Enter the url of the ops tools you want to connect to');
+            if (!url) {
+                return;
+            }
             const apiKey: string = await UserInputUtil.showInputBox('Enter the api key of the ops tools you want to connect to');
+            if (!apiKey) {
+                return;
+            }
 
             try {
                 const api: string = url.replace(/\/$/, '') + GET_ALL_COMPONENTS;
                 const response: any = await Axios.get(api, {
                     headers: { Authorization: `Bearer ${apiKey}` }
                 });
 
+                try {
+                    await keytar.setPassword('blockchain-vscode-ext', url, apiKey);
+                } catch (error) {
+                    outputAdapter.log(LogType.ERROR, `Unable to store API key securely in your keychain: ${error.message}`, `Unable to store API key securely in your keychain: ${error.toString()}`);
+                    return;
```

Comment:
you need to throw this error so that it gets caught at the bottom so it handles it correctly if it was called from adding an environment. If you just return then add environment will assume it was successful